src/code/z_skelanime.c

```diff
@@ -1,176 +1,1670 @@
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A2E70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A32EC.s")
+s16 func_800A2DBC(u32 animationSeg) {
+    AnimationHeader* animation = SEGMENTED_TO_VIRTUAL(animationSeg);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A32F4.s")
+    return animation->unk_02;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3310.s")
+s16 func_800A2DF4(u32 animationSeg) {
+    AnimationHeader* animation = SEGMENTED_TO_VIRTUAL(animationSeg);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3334.s")
+    return animation->frameCount;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A336C.s")
+s16 func_800A2E2C(u32 animationSeg) {
+    AnimationHeader* animation = SEGMENTED_TO_VIRTUAL(animationSeg);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3478.s")
+    return (s16)(animation->frameCount - 1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A34DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3548.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A35B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3620.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3678.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A36A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3714.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3770.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A37F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3874.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A390C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A39AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3B8C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3BC0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3BE4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3C9C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3D70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3E0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A3EE8.s")
+// Very large update function
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/func_800A2E70.s")
 
+void func_800A32EC(u16* arg0) {
+    *arg0 = 0;
+}
+
+void func_800A32F4(s32 arg0) {
+    D_801600B0 *= 2;
+}
+
+void func_800A3310(void* arg0) {
+    u32* D_8012A480Ptr = &D_8012A480;
+
+    *D_8012A480Ptr |= D_801600B0;
+}
+
+AnimationEntry* SkelAnime_NextEntry(AnimationContext* animationCtx, AnimationType type) {
+    AnimationEntry* entry;
+    s16 index = animationCtx->animationCount;
+
+    if (index >= 0x32) {
```

Comment:
Is 0x32 the maximum capacity of the animation entries? If so, I would suggest replacing 0x32 with a #define.

Reply:
Done, using ANIMATION_ENTRY_MAX now